cpp/src/arrow/filesystem/gcsfs.cc

```diff
@@ -102,6 +109,42 @@ class GcsInputStream : public arrow::io::InputStream {
   mutable gcs::ObjectReadStream stream_;
 };
 
+class GcsOutputStream : public arrow::io::OutputStream {
+ public:
+  explicit GcsOutputStream(gcs::ObjectWriteStream stream) : stream_(std::move(stream)) {}
+  ~GcsOutputStream() override = default;
+
+  Status Close() override {
+    stream_.Close();
+    return internal::ToArrowStatus(stream_.last_status());
```

Comment:
Does `last_status` also clear the error status or is it sticky? If it's sticky, then a failed `Write` would also return an error when calling `Close`?

Reply:
It is sticky.  And yes, a failed `Write()` will make subsequent `Close()` fail.  It is unadvisable to finalize a stream that failed, you don't know what is its state.